src/v/model/record.h

```diff
@@ -620,7 +629,7 @@ struct batch_identity {
     timestamp max_timestamp;
     bool is_transactional{false};
 
-    bool is_idempotent() const { return pid.id >= 0; }
+    bool is_idempotent() const { return pid.id != no_producer_id; }
```

Comment:
is this logic same in (kafka) client? for example (id, epoch) = (-2, 0) was earlier not idempotent but it is now.. it probably is but just wanted to double check.